docs/Userguide_HPO.rst

```diff
@@ -0,0 +1,110 @@
+Hyperparameter Optimization
+===========================
+
+Hyperparameter optimization is very easy to parallelize as each trial (unique set of hyperparameters) are
+independant of each other. The easiest way is to launch as many jobs as possible each trying a different
+set of hyperparameters and reporting their results back to a synchronized location (database).
+
+You will need to estimate how much resources your training requires and update the provided example to fit.
+In the example below we use 100 tasks with each 4 CPU cores and one GPU.
+Each task will run 4 training in parallel on the same GPU to maximize its utilization.
+
+This means there could be 400 set of hyperparameters being worked on in parallel across 100 GPUs.
+
+The easiest way to run an hyperparameter search on the cluster is simply to use a job array,
+which will launch the same job n times. Your HPO library will generate different parameters to try
+for each instances.
+
+.. code-block
+
+    sbatch --array=0-100 --gres=gpu:1 --cpus-per-gpu=2 --mem-per-gpu=16G scripts/hpo_launcher.sh train.py
```

Comment:
`--array=0-100` means 101 tasks, isn't it?

Reply:
ah right

Reply:
fixed